Issue with Mac Mini

The old version worked perfectly on my Mac mini with Catalina. After the update the monitor is not turned off anymore. I see the motion in the Display debug information UI and don't get any error messages, but the monitor stays on. The funny thing is that it shows the off time correctly in the UI even though the monitor is always on. Any idea what could be wrong?
If I try "pmset displaysleepnow" direct in terminal, my screen goes to sleep successfully.

My changes as it used to work on the Mac with the Old version:

``
const NodeHelper = require("node_helper");
const exec = require("child_process").exec;


module.exports = NodeHelper.create({
	start: function () {
		this.started = false;
		this.isMonitorOn(function(result) {
			console.log("MMM-MotionDetector: monitor on " + result);
		});
	},

	activateMonitor: function () {
		this.isMonitorOn(function(result) {
			if (!result) {
				exec('caffeinate -u -t 1', function(err, out, code) {
					if (err) {
						console.error("MMM-MotionDetector: error activating monitor: " + code);
					} else {
						console.log("MMM-MotionDetector: monitor has been activated");
					}
				});
			}
		});
		this.started = false;
	},

	deactivateMonitor: function () {
		this.isMonitorOn(function(result) {
			if (result) {
				exec('pmset displaysleepnow', function(err, out, code) {
					if (err) {
						console.error("MMM-MotionDetector: error deactivating monitor: " + code);
					} else {
						console.log("MMM-MotionDetector: monitor has been deactivated");
					}
				});
			}
		});
		this.started = false;
	},

	isMonitorOn: function(resultCallback) {
		exec('pmset -g powerstate IODisplayWrangler | tail -1 | cut -c29', function(err, out, code) {
			if (err) {
				console.error("MMM-MotionDetector: error calling monitor status: " + code);
				return;
			}

			console.log("MMM-MotionDetector: monitor " + out);
			resultCallback(out.includes("=1"));
		});
	},

	// Subclass socketNotificationReceived received.
	socketNotificationReceived: function (notification, payload) {
		if (notification === "MOTION_DETECTED" && this.started === false) {
			console.log("MMM-MotionDetector: MOTION_DETECTED, score " + payload.score);
			this.started = true;
			this.activateMonitor();
		}
		if (notification === "DEACTIVATE_MONITOR" && this.started === false) {
			console.log("MMM-MotionDetector: DEACTIVATE_MONITOR, percentage off: " + payload.percentageOff);
			this.started = true;
			this.deactivateMonitor();
		}
	}
});
`

So I updated your Module to the latest version. Now my screen didn't go to sleep I I got this error message:

`[2020-12-09 22:17:58.647] [LOG]    Launching application.
[2020-12-09 22:17:58.940] [LOG]    Whoops! There was an uncaught exception...
[2020-12-09 22:17:58.940] [ERROR]  TypeError: Log.info is not a function
    at /Users/sladak/MagicMirror/modules/MMM-MotionDetector/node_helper.js:47:11
    at ChildProcess.exithandler (child_process.js:294:7)
    at ChildProcess.emit (events.js:200:13)
    at maybeClose (internal/child_process.js:1021:16)
    at Socket.<anonymous> (internal/child_process.js:430:11)
    at Socket.emit (events.js:200:13)
    at Pipe.<anonymous> (net.js:586:12)
[2020-12-09 22:17:58.944] [LOG]    MagicMirror will not quit, but it might be a good idea to check why this happened. Maybe no internet connection?
[2020-12-09 22:17:58.944] [LOG]    If you think this really is an issue, please open an issue on GitHub: https://github.com/MichMich/MagicMirror/issues
[2020-12-09 22:18:00.170] [LOG]    Whoops! There was an uncaught exception...
[2020-12-09 22:18:00.171] [ERROR]  TypeError: Log.info is not a function
    at Class.socketNotificationReceived (/Users/sladak/MagicMirror/modules/MMM-MotionDetector/node_helper.js:54:11)
    at Socket.<anonymous> (/Users/sladak/MagicMirror/js/node_helper.js:113:11)
    at Socket.emit (events.js:200:13)
    at /Users/sladak/MagicMirror/node_modules/socket.io/lib/socket.js:528:12
    at processTicksAndRejections (internal/process/task_queues.js:82:9)
[2020-12-09 22:18:00.171] [LOG]    MagicMirror will not quit, but it might be a good idea to check why this happened. Maybe no internet connection?
`

--- a/LICENSE
+++ b/LICENSE
@@ -1,6 +1,6 @@
 MIT License
 
-Copyright (c) 2019 Veeck
+Copyright (c) 2020 rejas
 
 Permission is hereby granted, free of charge, to any person obtaining a copy
 of this software and associated documentation files (the "Software"), to deal

--- a/README.md
+++ b/README.md
@@ -85,6 +85,12 @@ If you want to use the wired PI-camera follow these steps provided by [@rev138](
 
 Another tutorial on how to enable the PI-camera in the browser can be found [in this blog post](https://reprage.com/post/pi-camera-module-in-the-browser).
 
+## Mac support
+
+Thanks to [3vidar](https://github.com/3vidar) you can also use this module on a Mac. For now the code only resides in the branch called "mac" in this git repo.
+
+I hope to find the time to generalize it and have it in the mainline too.
+
 ## Using the module
 
 To use this module, add it to the modules array in the `config/config.js` file:

--- a/diff-cam-engine.js
+++ b/diff-cam-engine.js
@@ -27,6 +27,10 @@ const DiffCamEngine = (function () {
 
   let coords;
 
+  /**
+   *
+   * @param options
+   */
   function init(options) {
     // sanity check
     if (!options) {
@@ -86,15 +90,12 @@ const DiffCamEngine = (function () {
     if (navigator.mediaDevices.getUserMedia === undefined) {
       navigator.mediaDevices.getUserMedia = function (constraints) {
         // First get a hold of the legacy getUserMedia, if present
-        const getUserMedia =
-          navigator.webkitGetUserMedia || navigator.mozGetUserMedia;
+        const getUserMedia = navigator.webkitGetUserMedia || navigator.mozGetUserMedia;
 
         // Some browsers just don't implement it - return a rejected promise with an error
         // to keep a consistent interface
         if (!getUserMedia) {
-          return Promise.reject(
-            new Error("getUserMedia is not implemented in this browser")
-          );
+          return Promise.reject(new Error("getUserMedia is not implemented in this browser"));
         }
 
         // Otherwise, wrap the call to the old navigator.getUserMedia with a Promise
@@ -123,6 +124,9 @@ const DiffCamEngine = (function () {
       });
   }
 
+  /**
+   *
+   */
   function start() {
     if (!stream) {
       throw "Cannot start after init fail";
@@ -132,12 +136,18 @@ const DiffCamEngine = (function () {
     video.addEventListener("canplay", startComplete);
   }
 
+  /**
+   *
+   */
   function startComplete() {
     video.removeEventListener("canplay", startComplete);
     captureInterval = setInterval(capture, captureIntervalTime);
     startCompleteCallback();
   }
 
+  /**
+   *
+   */
   function stop() {
     clearInterval(captureInterval);
     video.src = "";
@@ -145,15 +155,13 @@ const DiffCamEngine = (function () {
     isReadyToDiff = false;
   }
 
+  /**
+   *
+   */
   function capture() {
     // save a full-sized copy of capture
     captureContext.drawImage(video, 0, 0, captureWidth, captureHeight);
-    const captureImageData = captureContext.getImageData(
-      0,
-      0,
-      captureWidth,
-      captureHeight
-    );
+    const captureImageData = captureContext.getImageData(0, 0, captureWidth, captureHeight);
 
     // diff current capture over previous capture, leftover from last time
     diffContext.globalCompositeOperation = "difference";
@@ -184,7 +192,7 @@ const DiffCamEngine = (function () {
         },
         checkMotionPixel: function (x, y) {
           return checkMotionPixel(this.motionPixels, x, y);
-        }
+        },
       });
     }
 
@@ -194,6 +202,11 @@ const DiffCamEngine = (function () {
     isReadyToDiff = true;
   }
 
+  /**
+   *
+   * @param diffImageData
+   * @returns {{score: number, motionBox: (*|{x: {min: *, max: *}, y: {min: *, max: *}}|undefined), motionPixels: *[]}}
+   */
   function processDiff(diffImageData) {
     const rgba = diffImageData.data;
 
@@ -217,11 +230,7 @@ const DiffCamEngine = (function () {
         }
 
         if (includeMotionPixels) {
-          motionPixels = calculateMotionPixels(
-            motionPixels,
-            coords.x,
-            coords.y
-          );
+          motionPixels = calculateMotionPixels(motionPixels, coords.x, coords.y);
         }
       }
     }
@@ -229,22 +238,34 @@ const DiffCamEngine = (function () {
     return {
       score: score,
       motionBox: score > scoreThreshold ? motionBox : undefined,
-      motionPixels: motionPixels
+      motionPixels: motionPixels,
     };
   }
 
+  /**
+   *
+   * @param pixelIndex
+   * @returns {{x: number, y: number}}
+   */
   function calculateCoordinates(pixelIndex) {
     return {
       x: pixelIndex % diffWidth,
-      y: Math.floor(pixelIndex / diffWidth)
+      y: Math.floor(pixelIndex / diffWidth),
     };
   }
 
+  /**
+   *
+   * @param currentMotionBox
+   * @param x
+   * @param y
+   * @returns {*|{x: {min: *, max: *}, y: {min: *, max: *}}}
+   */
   function calculateMotionBox(currentMotionBox, x, y) {
     // init motion box on demand
     let motionBox = currentMotionBox || {
       x: { min: coords.x, max: x },
-      y: { min: coords.y, max: y }
+      y: { min: coords.y, max: y },
     };
 
     motionBox.x.min = Math.min(motionBox.x.min, x);
@@ -255,6 +276,13 @@ const DiffCamEngine = (function () {
     return motionBox;
   }
 
+  /**
+   *
+   * @param motionPixels
+   * @param x
+   * @param y
+   * @returns {*}
+   */
   function calculateMotionPixels(motionPixels, x, y) {
     motionPixels[x] = motionPixels[x] || [];
     motionPixels[x][y] = true;
@@ -262,28 +290,56 @@ const DiffCamEngine = (function () {
     return motionPixels;
   }
 
+  /**
+   *
+   * @param captureImageData
+   * @returns {string | null}
+   */
   function getCaptureUrl(captureImageData) {
     // may as well borrow captureCanvas
     captureContext.putImageData(captureImageData, 0, 0);
     return captureCanvas.toDataURL();
   }
 
+  /**
+   *
+   * @param motionPixels
+   * @param x
+   * @param y
+   * @returns {*}
+   */
   function checkMotionPixel(motionPixels, x, y) {
     return motionPixels && motionPixels[x] && motionPixels[x][y];
   }
 
+  /**
+   *
+   * @returns {*}
+   */
   function getPixelDiffThreshold() {
     return pixelDiffThreshold;
   }
 
+  /**
+   *
+   * @param val
+   */
   function setPixelDiffThreshold(val) {
     pixelDiffThreshold = val;
   }
 
+  /**
+   *
+   * @returns {*}
+   */
   function getScoreThreshold() {
     return scoreThreshold;
   }
 
+  /**
+   *
+   * @param val
+   */
   function setScoreThreshold(val) {
     scoreThreshold = val;
   }
@@ -298,6 +354,6 @@ const DiffCamEngine = (function () {
     // public functions
     init: init,
     start: start,
-    stop: stop
+    stop: stop,
   };
 })();

--- a/node_helper.js
+++ b/node_helper.js
@@ -3,12 +3,18 @@ const exec = require("child_process").exec;
 const Log = require("../../js/logger");
 
 module.exports = NodeHelper.create({
+  /**
+   *
+   */
   start: function () {
     this.isMonitorOn(function (result) {
       Log.info("MMM-MotionDetector: monitor is " + (result ? "ON" : "OFF") + ".");
     });
   },
 
+  /**
+   *
+   */
   activateMonitor: function () {
     this.isMonitorOn(function (result) {
       if (!result) {
@@ -23,6 +29,9 @@ module.exports = NodeHelper.create({
     });
   },
 
+  /**
+   *
+   */
   deactivateMonitor: function () {
     this.isMonitorOn(function (result) {
       if (result) {
@@ -37,6 +46,10 @@ module.exports = NodeHelper.create({
     });
   },
 
+  /**
+   *
+   * @param resultCallback
+   */
   isMonitorOn: function (resultCallback) {
     exec("vcgencmd display_power", function (err, out, code) {
       if (err) {
@@ -49,6 +62,11 @@ module.exports = NodeHelper.create({
     });
   },
 
+  /**
+   *
+   * @param notification
+   * @param payload
+   */
   socketNotificationReceived: function (notification, payload) {
     if (notification === "ACTIVATE_MONITOR") {
       Log.info("MMM-MotionDetector: activating monitor.");

